### ISSUE TITLE
Service brought up by docker-compose seems to have slower response time than docker run

### ISSUE THREAD

**Issue by mundimduy2506** [docker/compose#5085]
I have a service call HealthCheck to verify database connection.
I use **docker-compose up** (with docker-compose.yml housing all environments variables) vs **docker run** (I must manually pass env variables in the command) to bring up the docker container but there is an observation:
If I hit this call
http://my-ip:8081/api/v1/HealthCheck

Service brought by docker-compose up: it takes around 2257 ms to return very call
while another with docker run: it takes < 1 second to give me the result

Docker-compose version: 1.8
Docker-engine version: 1.12.3

The time to start container seems similar in both ways
Any ideas why there is the slowness with docker-compose?

**Comment by shin-** [comment]
How many times have you made the comparison? Response times over the network can vary a lot depending on a number of independent factors. If the average over 100, 1000 calls supports your observation, it's something we can look into.

Also: the version of Compose you are using is over 1 year old, you should consider upgrading.

**Comment by mundimduy2506** [comment]
yes, we have docker-compose upgraded to 1.10 last week but since there was incompatible issue so we have to rollback. 
This is another issue, out of topic but I would like to mention
when we upgraded to 1.10 
**Docker-compose up** works, all containers were created but database was not connected. 

"IMSApiSqlException: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 40 - Could not open a connection to SQL Server)"

As the same time, db is connect if running with **docker run**

I searched around and couldnt know why coz Docker-compose 1.10 is compatible with docker-engine 1.12+
https://github.com/docker/compose/releases/tag/1.11.0

Since we have run docker-compose up for a long time so folks decided to rollback.
